plugins/modules/podman_generate_systemd.py

```diff
@@ -0,0 +1,525 @@
+            module.fail_json(
+                msg=f'Restart policy requested is "{restart_policy}"'
+                f',  but must be one of: {RESTART_POLICY_CHOICES}'
+            )
+        # Else add the restart policy to options
+        if restart_policy == 'no-restart':
+            restart_policy = 'no'
+        command_options.append(f'--restart-policy={restart_policy}')
+
+    #  Restart-sec option (only for Podman 4.0.0 and above)
+    restart_sec = module.params.get('restart_sec')
+    if restart_sec:
+        command_options.append(f'--restart-sec={restart_sec}')
+
+    #  Start-timeout option (only for Podman 4.0.0 and above)
+    start_timeout = module.params.get('start_timeout')
+    if start_timeout:
+        command_options.append(f'--start-timeout={start_timeout}')
+
+    #  Stop-timeout option
+    stop_timeout = module.params.get('stop_timeout')
+    if stop_timeout:
+        command_options.append(f'--stop-timeout={stop_timeout}')
+
+    #  Use container name(s) option
+    if module.params.get('use_names'):
+        command_options.append('--name')
+
+    #  Container-prefix option
+    container_prefix = module.params.get('container_prefix')
+    if container_prefix:
+        command_options.append(f'--container-prefix={container_prefix}')
+
+    #  Pod-prefix option
+    pod_prefix = module.params.get('pod_prefix')
+    if pod_prefix:
+        command_options.append(f'--pod-prefix={pod_prefix}')
+
+    #  Separator option
+    separator = module.params.get('separator')
+    if separator:
+        command_options.append(f'--separator={separator}')
+
+    #  No-header option
+    if module.params.get('no_header'):
+        command_options.append('--no-header')
+
+    #  After option (only for Podman 4.0.0 and above)
+    after = module.params.get('after')
+    if after:
+        for item in after:
+            command_options.append(f'--after={item}')
+
+    #  Wants option (only for Podman 4.0.0 and above)
+    wants = module.params.get('wants')
+    if wants:
+        for item in wants:
+            command_options.append(f'--wants={item}')
+
+    #  Requires option (only for Podman 4.0.0 and above)
+    requires = module.params.get('requires')
+    if requires:
+        for item in requires:
+            command_options.append(f'--requires={item}')
+
+    # Environment variables (only for Podman 4.3.0 and above)
+    environment_variables = module.params.get('env')
+    if environment_variables:
+        for env_var_name, env_var_value in environment_variables.items():
+            command_options.append(
+                f"-e='{env_var_name}={env_var_value}'",
+            )
+
+    #  Full command, with option include
+    command_options.extend(['--format', 'json'])
+    command = [
+        module.params.get('executable'), 'generate', 'systemd',
+        *command_options,
+        module.params['name'],
+    ]
+    command_str = ' '.join(command)
```

Comment:
Done.

Reply:
So, using `shlex.join()` generate a weird command with the `-e='ENV_VAR=value` option. And it's seems not possible to avoid it.

As `command_str` is only used for module return and it's made with user given options, maybe I can come back to `' '.join(command)`

Reply:
Ack